### PR TITLE
element call: Remove `E2EEenabled` flag

### PR DESCRIPTION
**Pull request by toger5**
This is a deprecated flag for element call using livekit. It was used to enable/disable matrix signalling event encryption. This is not relevant for embedded mode at all since there the hosting client is taking care of encryption.

- [ ] Public API changes documented in changelogs (optional)

Signed-off-by:

### CODE REVIEW

**Review by bnjbvr — CHANGES_REQUESTED** [review]

File: crates/matrix-sdk/src/widget/settings/element_call.rs [hunk 1]
```diff
@@ -380,9 +372,9 @@ mod tests {
                 EncryptionSystem::Unencrypted,
             )));
             let query_set = get_query_sets(&Url::parse(&url).unwrap()).unwrap().1;
-            let expected_elements = ("enableE2EE".to_owned(), "false".to_owned());
+            let expected_elements = ("perParticipantE2EE".to_owned(), "true".to_owned());
```
[the remark this answers is not in the document]
Comment (toger5) [reply crates/matrix-sdk/src/widget/settings/element_call.rs:375]: Sorry the comment was wrong. I was misinformed at the point of writing. (The core part: that this is deprecated) was correct. The reason why wasnt.

[the remark this answers is not in the document]
Comment (toger5) [reply crates/matrix-sdk/src/widget/settings/element_call.rs:375]: But thanks for flaggin the test. Its not good as is. (I at least need to update the assert log.

File: bindings/matrix-sdk-ffi/src/widget.rs [hunk 2]
```diff
@@ -113,17 +113,16 @@ pub async fn generate_webview_url(
...
 /// Defines if a call is encrypted and which encryption system should be used.
 ///
-/// This controls the url parameters: `enableE2EE`, `perParticipantE2EE`,
-/// `password`.
+/// This controls the url parameters: `perParticipantE2EE`, `password`.
 #[derive(uniffi::Enum, Clone)]
 pub enum EncryptionSystem {
     /// Equivalent to the element call url parameter: `enableE2EE=false`
     Unencrypted,
     /// Equivalent to the element call url parameters:
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/widget.rs:121]: (sorry about the nitpick, trying to be consistent with the previous comments :grin:)

```suggestion
    /// Equivalent to the element call url parameter:
```

File: bindings/matrix-sdk-ffi/src/widget.rs [hunk 3]
```diff
@@ -113,17 +113,16 @@ pub async fn generate_webview_url(
...
-    /// `enableE2EE=true&perParticipantE2EE=true`
+    /// `perParticipantE2EE=true`
     PerParticipantKeys,
     /// Equivalent to the element call url parameters:
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/widget.rs:124]:
```suggestion
    /// Equivalent to the element call url parameter:
```

**Review by bnjbvr — APPROVED** [review]
Thanks!

One small question: we're never translating from a raw URL with encoded query parameters to an `EncryptionSystem`, are we? If so, is there a test showing which we prefer between `PerParticipantKeys` and `SharedSecret` when both are set?

**Comment by toger5** [comment]
We only do it in the opposite direction: `EncryptionSystem` -> raw url and never the other way around.

The other way around is only done in element call (the webapp/widget). There we have a clear order of what we use. (its outside of the rust sdk logic. Even this is already not a perfect fit to have in the rust sdk but otherwise ios and android would need to do this by themselves and it is very encapsulated so the rust sdk is a good place to make it reliable/patchable and correct on both systems)

**Pull request merged**